tests/functional/Ticket.php

```diff
@@ -560,6 +560,383 @@ public function testUpdateTicketWithActors(array $actors_input, array $expected_
         $this->checkActors($ticket, $expected_actors);
     }
 
+    /**
+     * Test that ticket anonymous users notifications are successfully updated.
+     *
+     * This method tests the functionality of updating ticket notifications for anonymous users.
+     * It creates a ticket with different actors (requester, observer, assign) and enables notifications.
+     * It then adds a follow-up to the ticket to trigger notifications and checks the notification queue.
+     * It updates the anonymous users to disable their notifications and checks the notification queue again.
+     * It then adds a second anonymous requester to the ticket and checks the notification queue again.
+     *
+     * @return void
+     */
+    public function testUpdateActorsNotification(): void
```

Comment:
suggestion applied

Reply:
I think what @cedric-anne meant is that checking the actors are correct after the ticket was updated is probably enough and we don't need to tests the notifications themselves here.

The idea is to keep the test lightweight and focus and the direct issue that we are trying to fix (= the inability the disable notification for an anonymous actor).

On another note, a good way to ensure a test case is relevant is to run it without its associated fix.
For example, I've checked out your new test case on my GLPI but I didn't apply the other changes :

![image](https://github.com/glpi-project/glpi/assets/42734840/3e144988-6872-4e6f-aaf0-08bfa3820a36)

If the test was relevant to the original issue then it would fail here since I didn't apply the fix ;)

Could you look into it ?
Also it would be perfect if the tests also covered the special case where we have more than one anonymous requester :)

Reply:
The test pass without applying the changes because you didn't run the test command on the right function. the function to test is `testUpdateActorsNotification` :)
And for the special case where we have more than one anonymous requester, it's already been tested [here](https://github.com/glpi-project/glpi/blob/5cf9748d440c054462dde97dfd61081466035bbb/tests/functional/Ticket.php#L749C9-L774C11).

Reply:
My bad, sorry about that.
Test is good 👍

Reply:
This test seems too complex (there is not need to check that notifications are sent, call to `$this->checkActors()` is enough).
Also, please add a case that contains more than one email actor for a given role, for instance, a test that creates a ticket with 3 email observers, and that would update the second one.

You could use a data provider method to be able to add many tests cases withour having to rewrite too many times the same logic.